src/steelseries_sonar_py/sonar.py

```diff
@@ -16,19 +17,47 @@ class Sonar:
         "coreProps.json",
     )
 
-    def __init__(self, streamer_mode=False, app_data_path=None):
+    def __init__(self, app_data_path=None, streamer_mode=None):
         requests.packages.urllib3.disable_warnings()
 
-        self.streamer_mode = streamer_mode
-        if self.streamer_mode:
-            self.volume_path = "/volumeSettings/streamer/monitoring"
-
         if app_data_path is not None:
             self.app_data_path = app_data_path
 
         self.load_base_url()
         self.load_server_address()
 
+        if streamer_mode is None:
+            # get the streamer mode using a request to the server
+            self.streamer_mode = Sonar.is_streamer_mode(self)
```

Comment:
The function is now inside the class, so you can call it like this: `self.streamer_mode = self.is_streamer_mode()`

Reply:
Yes, self. is simpler.